django/urls/resolvers.py

```diff
@@ -429,7 +430,10 @@ def _check_custom_error_handlers(self):
                 messages.append(Error(msg, hint=str(e), id='urls.E008'))
                 continue
             signature = inspect.signature(handler)
-            args = [None] * num_parameters
+            if signature.parameters.get('self'):
+                args = [None] * 2  # Class based views always take two arguements, (self, request).
+            else:
+                args = [None] * num_parameters
```

Comment:
I think you want to allow an extra parameter when it's a class based view

In other words

```
# Class based views always take two arguements, (self, request).
```

Is not not true for some of these handlers.


```suggestion
            if signature.parameters.get('self'):
                num_parameters += 1
            args = [None] * num_parameters
```

Reply:
Thank you for your suggestion!
Simply added +1 to the num_parameters will make the value of num_parameters for handler404, handler403 and handler400 as 3 which is not correct. 
https://github.com/django/django/blob/8d7085e0fd004af5431389f3d903aba6220d7957/django/urls/resolvers.py#L421

I can change it as `if signature.parameters.get('self') and status_code==500: ` but that would be the greatest solution in my opinion.

Reply:
> You should add tests for CBV handlers as well

Yes I will do that right away!

Reply:
Yes I love SwissBorg tokens